python/social_bot/tasks.py

```diff
@@ -257,15 +268,23 @@ def __init__(self,
             distraction_list.append(goal_name)
         self._distraction_list = distraction_list
         self._object_list = distraction_list
-        self._move_goal_during_episode = move_goal_during_episode
-        self._success_with_angle_requirement = success_with_angle_requirement
-        self._additional_observation_list = additional_observation_list
         if goal_name and goal_name not in distraction_list:
             self._object_list.append(goal_name)
         self._goals = self._object_list
-        self._pos_list = list(itertools.product(range(-5, 5), range(-5, 5)))
+        self._move_goal_during_episode = move_goal_during_episode
+        self._success_with_angle_requirement = success_with_angle_requirement
+        if not additional_observation_list:
+            additional_observation_list = self._object_list
```

Comment:
@le-horizon Why did you make this change? This change the observation. Several tasks if alf cannot run because of this change (https://github.com/HorizonRobotics/alf/pull/334/checks?check_run_id=445009810)

Reply:
Hmm, this is so that full internal state input would include all objects in the world.
I'll check tmr how to fix the tests.  It's probably also a good idea to make alf tests run on socialbot PR's.

Reply:
@emailweixu I tested without these two lines, still getting the same error as with the lines:

`  File "/home/lezhao/alf/alf/algorithms/misc_algorithm.py", line 163, in train_step
    add_batch()
  File "/home/lezhao/alf/alf/algorithms/misc_algorithm.py", line 160, in add_batch
    self._buffer.add_batch(feature_reshaped_tran)
  File "/home/lezhao/alf/alf/utils/data_buffer.py", line 94, in add_batch
    indices, tf.stop_gradient(bat[-n:])), self._buffer, batch)
  File "/home/lezhao/venv/lib/python3.5/site-packages/tensorflow_core/python/util/nest.py", line 535, in map_structure
    structure[0], [func(*x) for x in entries],
  File "/home/lezhao/venv/lib/python3.5/site-packages/tensorflow_core/python/util/nest.py", line 535, in <listcomp>
    structure[0], [func(*x) for x in entries],
  File "/home/lezhao/alf/alf/utils/data_buffer.py", line 94, in <lambda>
    indices, tf.stop_gradient(bat[-n:])), self._buffer, batch)
  File "/home/lezhao/venv/lib/python3.5/site-packages/tensorflow_core/python/ops/resource_variable_ops.py", line 1180, in scatter_nd_update
    name=name))
  File "/home/lezhao/venv/lib/python3.5/site-packages/tensorflow_core/python/ops/gen_state_ops.py", line 680, in resource_scatter_nd_update
    _six.raise_from(_core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: Must have updates.shape = indices.shape[:batch_dim] + params_shape[slice_dim:], got updates.shape: [1,4,23], indices.shape: [1,1], params_shape: [100,3,23], slice_dim: 1, and batch_dim: 1 [Op:ResourceScatterNdUpdate]
  In call to configurable 'SyncOffPolicyDriver' (<function SyncOffPolicyDriver.__init__ at 0x7fb4abb8c2f0>)
  In call to configurable 'train_eval' (<function train_eval at 0x7fb4abb71d90>)
`

Is this what you saw?  Might be something else affecting it.